cmd/podman/kube/play.go

```diff
@@ -265,7 +271,33 @@ func play(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	return kubeplay(reader)
+	if err = kubeplay(reader); err != nil {
+		return err
+	}
+
+	if playOptions.Wait {
+		fmt.Println("Use ctrl+c to clean up")
+		// Stop the the shutdown signal handler so we can actually clean up after a SIGTERM or interrupt
+		if err := shutdown.Stop(); err != nil {
```

Comment:
There's a race condition on receiving SIGTERM.  Can the --wait handler be installed at an earlier point?

Reply:
Where is the race happening? I don't think I can move it any earlier, we need `kubeplay(reader)` to be called before we handle the --wait.

Reply:
> Where is the race happening? 

In between calling `kubeplay(...)` and here.

> I don't think I can move it any earlier, we need `kubeplay(reader)` to be called before we handle the --wait.

Can you elaborate?

Reply:
Can you please explain how exactly the race could happen, I am not seeing it.

`kubeplay(reader)` is what reads the kube yaml and creates the containers, pods, and volumes - that is why it needs to be called first before we check for the --wait flag and wait on a ctrl+c to clean up if it is set to true. If I do it after checking for --wait, we will just be waiting for an interrupt without creating any of the workloads from the kube yaml file.
The code can be changes to call `kubeplay(reader)` twice, once inside the `if playOptions.Wait{` check and then an else to call it again.

Reply:
Imagine the signal comes in after `kubeplay()` but before installing the signal handler; that's the race window. You may need more than one goroutine to fix the race.

Reply:
Fixed